### ISSUE TITLE
Remove '#line 1' from flex generated files to avoid bug in nvcc with CMake + Ninja + CUDA builds of Trilinos

### ISSUE THREAD

**Issue by bartlettroscoe** [gsjaardema/seacas#97]
There is a strange defect with `nvcc` when calling through Ninja as generated by CMake described in trilinos/Trilinos#2359 that causes `nvcc` to spit out incorrect dependencies for all line in source files of the form `#line 1 <some-file>`.  This results in unnecessary rebuilds of a bunch of files when ever `ninja` is called to build.  A defect report has been filed with NVIDA but the workaround is to just remove the lines `#line 1 <some-file>` from all of the `flex` generated SEACAS source files.  In particular, with the patch:

```
diff --git a/packages/seacas/libraries/aprepro_lib/apr_scanner.cc b/packages/seacas/libraries/aprepro_lib/apr_scanner.cc
index 747470a..a2512cd 100644
--- a/packages/seacas/libraries/aprepro_lib/apr_scanner.cc
+++ b/packages/seacas/libraries/aprepro_lib/apr_scanner.cc
@@ -971,7 +971,6 @@ static yyconst flex_int16_t yy_rule_linenum[102] = {
 #define yymore() yymore_used_but_not_detected
 #define YY_MORE_ADJ 0
 #define YY_RESTORE_YY_MORE_OFFSET
-#line 1 "/scratch/gdsjaar/seacas/packages/seacas/libraries/aprepro_lib/aprepro.ll"
 /* -*- Mode: c++ -*- */
 #line 39 "/scratch/gdsjaar/seacas/packages/seacas/libraries/aprepro_lib/aprepro.ll"
```

or the patch

```
diff --git a/packages/seacas/libraries/aprepro_lib/apr_scanner.cc b/packages/seacas/libraries/aprepro_lib/apr_scanner.cc
index 747470a..5774992 100644
--- a/packages/seacas/libraries/aprepro_lib/apr_scanner.cc
+++ b/packages/seacas/libraries/aprepro_lib/apr_scanner.cc
@@ -971,7 +971,7 @@ static yyconst flex_int16_t yy_rule_linenum[102] = {
 #define yymore() yymore_used_but_not_detected
 #define YY_MORE_ADJ 0
 #define YY_RESTORE_YY_MORE_OFFSET
-#line 1 "/scratch/gdsjaar/seacas/packages/seacas/libraries/aprepro_lib/aprepro.ll"
+#line 2 "/scratch/gdsjaar/seacas/packages/seacas/libraries/aprepro_lib/aprepro.ll"
 /* -*- Mode: c++ -*- */
 #line 39 "/scratch/gdsjaar/seacas/packages/seacas/libraries/aprepro_lib/aprepro.ll"
```

both solve the problem.

I can change this in the snapshot of this file in Trilinos but it would just get overwritten the next time that SEACAS was snapshotted into Trilinos.

Is there any way to avoid this `#line 1` from getting into these generated files?

**Comment by bartlettroscoe** [comment]
@gsjaardema ,

I created the PR #98 for a short-term fix for this.  But the long term fix (other than getting an updated CUDA with a fixed nvcc) is to avoid having these generated files have `#line 1 <some-file>`.

What is the best long-term solution for avoiding `#line 1 ` lines?

**Comment by gsjaardema** [comment]
Fixed in 0020417.
